apps/mobile-wallet/src/features/transactionsDisplay/InputsOutputsLists.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  GENESIS_TIMESTAMP,
+  getTransactionDestinationAddresses,
+  getTransactionOriginAddresses,
+  isConfirmedTx,
+  selectPendingSentTransactionByHash,
+  UseTransactionProps
+} from '@alephium/shared'
+import { explorer as e } from '@alephium/web3'
+import { useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components/native'
+
+import AddressBadge from '~/components/AddressBadge'
+import AppText from '~/components/AppText'
+import PendingSentAddressBadge from '~/features/transactionsDisplay/PendingSentAddressBadge'
+import { useAppSelector } from '~/hooks/redux'
+
+interface InputsListProps extends UseTransactionProps {
+  tx: e.Transaction | e.PendingTransaction
+}
+
+export const TransactionOriginAddressesList = ({ tx, referenceAddress }: InputsListProps) => {
+  const { t } = useTranslation()
+  const pendingSentTx = useAppSelector((s) => selectPendingSentTransactionByHash(s, tx.hash))
+
+  const addresses = useMemo(() => getTransactionOriginAddresses({ tx, referenceAddress }), [tx, referenceAddress])
+
+  if (pendingSentTx) {
+    return <PendingSentAddressBadge txHash={tx.hash} direction="from" />
+  }
+
+  if (GENESIS_TIMESTAMP === (isConfirmedTx(tx) ? tx.timestamp : undefined)) {
```

Comment:
Nit: formulating the condition like this is a bit weird to me (the `undefined` threw me off a bit). Why not:

```ts
if (isConfirmedTx(tx) && tx.timestamp === GENESIS_TIMESTAMP) {...}
```